rocksdb_replicator/replicated_db.cpp

```diff
@@ -446,8 +447,13 @@ void RocksDBReplicator::ReplicatedDB::handleReplicateRequest(
     logMetric(kReplicatorLeaderSequenceNumbersBehind, seq_no - leaderSeqNum, db ->db_name_);
   }
 
-  // post the largest sequence number the Slave has committed
-  max_seq_no_acked_.post(seq_no);
+  // Post the largest sequence number the Slave has committed.
+  // If the request comes from an Observer, ignore the sequence number update.
+  if (request->__isset.role && request->role == ReplicaRole::OBSERVER) {
+    incCounter(kReplicatorHandleObserverRequests, 1, db->db_name_);
+  } else {
+    max_seq_no_acked_.post(seq_no);
```

Comment:
Do we want a stat for this case as well?

Reply:
We could, though this is the common/default case, so not as important/useful as the observer requests where we need more visibility